tests/functions.tests

```diff
@@ -146,3 +146,19 @@ Linearity (9) - linear recursive functions should be disallowed
 linfun f(x) {f(x)} f
 stderr : @.*cannot be linear.*
 exit : 1
+
+Pattern Matching (1):
+fun foo(_) match {| x -> x} foo
```

Comment:
As a matter of visual style I suggest having a space between `{` and `|` since otherwise it looks like `{|` is a single token - I might then expect to see a closing `|}` at the end.  (Links uses some other similar-looking tokens as parentheses such as `[| ... |]` for variants.)

Alternatively perhaps the initial "|" in a match block could be optional, or perhaps we could explicitly require "match braces" around match blocks, i.e. 

```
 fun foo(_) match {| x -> x |}
 fun ack(_,_) match {| (0, n) -> n + 1 | (m, 0) -> ack(m - 1, 1) | (m, n) -> ack(m - 1, ack(m, n - 1)) |} 
```

(In fact if we did this (by adding new tokens `{|`, `|}` then the `match` keyword would be redundant...)